posts/ngrx-facade-pattern.md

```diff
@@ -0,0 +1,137 @@
+# Facade Pattern with NgRx
+Over the last few years, I have seen many developers struggle with state management and understanding the [Redux Pattern](https://redux.js.org/understanding/thinking-in-redux/three-principles). By implementing the Facade Pattern, you can add a layer of abstraction over your state management and decouple the knowledge of stores from your component. In this blog, we will discuss the Facade Pattern with NgRx and the benefits of implementing them together.
```

Comment:
This combined with your closing sentences confuses me - the setup you have here leads me to believe that you're recommending the Facade Pattern as an abstraction for those who are confused by Redux/NgRx, but at the end you say that teams should only consider the Facade Pattern if they have a solid understanding of NgRx, not as an obfuscation.

Reply:
I will reword my conclusion, but I want to convey that you should not *skip* teaching devs about NgRx. If the service is written, for the most part they wouldn't have to interact with it. 

The abstraction is good to move fast for those that are unfamiliar. In the future not knowing how stores work will only become a hindrance for new intent/whenever you will have to write it yourself.

Reply:
Roger that. So abstracting the NgRx into a service functions to allow a newer dev to focus on what is component logic separately from what is NgRx logic, simplifying each while they look at each

Reply:
Yes, I rewrote my conclusion to try to make that more clear.